### PR TITLE
feat(limps): add tool filtering and planning skill docs

### PR DESCRIPTION
**Pull request by paulbreuler**
## Summary
- add tool filtering (allow/deny) for MCP tool registration
- expand README with TOC/features/env/transport/skills + Windows snippet
- add planning skill package and plan 0044 artifacts

## Changes
- `packages/limps/src/config.ts`: tool filtering config + validation
- `packages/limps/src/tools/index.ts`: filter tool registration
- `packages/limps/tests/*`: add allow/deny config tests
- `README.md`, `config.json.example`, `packages/limps-headless/README.md`: doc updates
- `skills/limps-planning/*`: planning skill package
- `plans/0044-mcp-server-chart-learnings/*`: new plan + agents

## Tests
- `npx vitest run tests/tool-registration.test.ts tests/config-validation.test.ts`
- `npm run format:check`
- `npm run lint`
- `npm run type-check`
- `npm run build`

## Code Review
- General review: Not run
- MCP/LLM review: Not run
- Commit review: Not run

## Breaking Changes
- None

## Notes / Risks
- Tool filtering can hide required tools if misconfigured; allowlist takes precedence.

## Plan / Agent
- Plan: 0044-mcp-server-chart-learnings
- Agent: Docs & README (`plans/0044-mcp-server-chart-learnings/agents/000_agent_docs.agent.md`)
- Agent: Tool Filtering (`plans/0044-mcp-server-chart-learnings/agents/001_agent_tool_filtering.agent.md`)
- Agent: Skills Packaging (`plans/0044-mcp-server-chart-learnings/agents/002_agent_skills_packaging.agent.md`)

### CODE REVIEW

File: packages/limps/tests/tool-registration.test.ts [hunk 1]
```diff
@@ -291,3 +291,33 @@ describe('tool-registration-no-duplicate', () => {
...
+    expect(filtered).not.toContain('process_doc');
+    expect(filtered).not.toContain('process_docs');
+  });
+
+  it('uses env vars when config missing', () => {
+    process.env.LIMPS_ALLOWED_TOOLS = 'list_docs,search_docs';
+    const filtered = filterToolNames(CORE_TOOL_NAMES, undefined, process.env);
+    expect(filtered).toEqual(['list_docs', 'search_docs']);
+  });
+});
```
[the remark this answers is not in the document]
Comment (paulbreuler) [reply packages/limps/tests/tool-registration.test.ts]: Added tests for empty allow/deny arrays and other edge cases. Commit 72b73c5.

File: packages/limps/src/tools/index.ts [hunk 2]
```diff
@@ -40,15 +41,107 @@ export const CORE_TOOL_NAMES = [
...
+};
+
 /**
  * Register all MCP tools with the server.
  *
  * @param server - MCP server instance
  * @param context - Tool context with db and config
  */
 export function registerTools(server: McpServer, context: ToolContext): void {
+  const enabledTools = new Set(filterToolNames(CORE_TOOL_NAMES, context.config.tools));
```
[the remark this answers is not in the document]
Comment (paulbreuler) [reply packages/limps/src/tools/index.ts]: Added warning when tool filtering results in zero enabled tools. Commit 72b73c5.

File: packages/limps/src/tools/index.ts [hunk 3]
```diff
@@ -40,15 +41,107 @@ export const CORE_TOOL_NAMES = [
...
+  const available = new Set(toolNames);
+  const allowlist = resolvedConfig.allowlist?.filter((name) => name.length > 0);
+  const denylist = resolvedConfig.denylist?.filter((name) => name.length > 0);
+
+  const unknownNames = [...(allowlist ?? []), ...(denylist ?? [])].filter(
+    (name) => !available.has(name)
+  );
+  if (unknownNames.length > 0) {
+    console.warn(`Unknown tool names in filter list: ${unknownNames.join(', ')}. Ignoring.`);
+  }
```
[the remark this answers is not in the document]
Comment (paulbreuler) [reply packages/limps/src/tools/index.ts]: Added test that captures console.warn for unknown tool names. Commit 72b73c5.

File: config.json.example [hunk 4]
```diff
@@ -3,6 +3,10 @@
   "docsPaths": ["/path/to/your/planning/documents"],
   "fileExtensions": [".md"],
   "dataPath": "./data",
+  "tools": {
+    "allowlist": ["list_docs", "search_docs"],
+    "denylist": ["process_doc"]
```
[the remark this answers is not in the document]
Comment (paulbreuler) [reply config.json.example]: Simplified config example to show allowlist only. Commit 72b73c5.
